core/src/main/java/org/jivesoftware/LoginDialog.java

```diff
@@ -277,6 +284,22 @@ protected XMPPTCPConnectionConfiguration retrieveConnectionConfiguration() {
             builder.setProxyInfo( proxyInfo );
         }
 
+        if (securityMode != ConnectionConfiguration.SecurityMode.disabled && !useOldSSL) {
```

Comment:
Indeed this two classes share a bunch of the similar code, tomorrow I will try to reduce some of its redundancy.